### PR TITLE
Bump aeson to 2.2

### PR DESCRIPTION
**Pull request by carbolymer**
# Description

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [x] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [x] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: hie.yaml
Comment (carbolymer) [inline hie.yaml:1]: TODO: restore this

File: libs/cardano-data/src/Data/ListMap.hs
Comment (carbolymer) [inline libs/cardano-data/src/Data/ListMap.hs:1]: Changes were copied from: https://hackage.haskell.org/package/aeson-2.2.1.0/docs/src/Data.Aeson.Types.FromJSON.html and https://hackage.haskell.org/package/aeson-2.2.1.0/docs/src/Data.Aeson.Types.ToJSON.html

**Comment by teodanciu** [comment]
This looks good to me, however I would rather @lehins  also approved this before merging, just in case he sees any potential problem or interference with the release we're trying to make. 
Adding his as reviewer

**Review by lehins — APPROVED** [review]
Thank you very much. Looks great!

**Pull request merged**